### PR TITLE
Improve bulk archive progress and mailbox links

### PR DESCRIPTION
**Pull request by elie222**
# User description
## Summary
- Improves assistant bulk archive feedback by showing sender details while the action is running.
- Adds external-link actions for sender rows so users can open results directly in their mailbox client.
- Adds a provider-aware sender search URL helper for Google and Microsoft providers.
- Adds URL tests for sender search link generation across providers.

---

# Generated description

Below is a concise technical summary of the changes proposed in this PR:
```mermaid
graph LR
ManageInboxResult_("ManageInboxResult"):::modified
getEmailSearchUrl_("getEmailSearchUrl"):::added
getManageInboxActionLabel_("getManageInboxActionLabel"):::modified
MessagePart_("MessagePart"):::modified
getInProgressManageInboxOutput_("getInProgressManageInboxOutput"):::added
buildSearchUrl_google_("buildSearchUrl (google)"):::added
buildSearchUrl_default_("buildSearchUrl (default)"):::added
getGmailSearchUrl_("getGmailSearchUrl"):::modified
ManageInboxResult_ -- "Now opens provider-specific search URLs for listed senders." --> getEmailSearchUrl_
ManageInboxResult_ -- "Includes in-progress state for dynamic action summary." --> getManageInboxActionLabel_
MessagePart_ -- "Renders in-progress ManageInboxResult for bulk sender archive." --> ManageInboxResult_
MessagePart_ -- "Synthesizes temporary output (senders, count) for progress display." --> getInProgressManageInboxOutput_
getEmailSearchUrl_ -- "Delegates to Google buildSearchUrl to encode 'from'." --> buildSearchUrl_google_
getEmailSearchUrl_ -- "Falls back to default Gmail-style buildSearchUrl when unknown." --> buildSearchUrl_default_
getGmailSearchUrl_ -- "Refactored to call provider buildSearchUrl for Gmail." --> buildSearchUrl_google_
classDef added stroke:#15AA7A
classDef removed stroke:#CD5270
classDef modified stroke:#EDAC4C
linkStyle default stroke:#CBD5E1,font-size:13px
```

Enhances the assistant's bulk archive feedback by displaying sender details during execution and introduces provider-aware external links to view search results in Gmail or Outlook. Updates the <code>ManageInboxResult</code> and <code>MessagePart</code> components to handle in-progress states and integrates a new URL helper for cross-provider mailbox navigation.
<table><tr><th>Topic</th><th>Details</th><tr><td><a href=https://baz.co/changes/elie222/inbox-zero/1629?tool=ast&topic=Assistant+UI%2FUX>Assistant UI/UX</a>
        </td><td>Improves the assistant's chat interface by showing active progress for bulk actions and rendering sender lists with external mailbox links.<details><summary>Modified files (2)</summary><ul><li>apps/web/components/assistant-chat/message-part.tsx</li>
<li>apps/web/components/assistant-chat/tools.tsx</li></ul></details><details><summary>Latest Contributors(0)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr></table></details></td></tr>
<tr><td><a href=https://baz.co/changes/elie222/inbox-zero/1629?tool=ast&topic=Mailbox+URL+Utils>Mailbox URL Utils</a>
        </td><td>Implements <code>getEmailSearchUrl</code> and provider configurations to generate deep links for Google and Microsoft mail clients, supported by new unit tests.<details><summary>Modified files (2)</summary><ul><li>apps/web/utils/url.test.ts</li>
<li>apps/web/utils/url.ts</li></ul></details><details><summary>Latest Contributors(0)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr></table></details></td></tr></table>
This pull request is reviewed by Baz. Review like a pro on <a href=https://baz.co/changes/elie222/inbox-zero/1629?tool=ast>(Baz)</a>.

### CODE REVIEW

File: apps/web/utils/url.ts [hunk 1]
```diff
@@ -78,9 +91,17 @@ export function getGmailUrl(
...
+}
+
+export function getEmailSearchUrl(
+  from: string,
+  emailAddress?: string | null,
+  provider?: string,
+) {
+  const config = getProviderConfig(provider);
+  return config.buildSearchUrl(from, emailAddress);
 }
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/utils/url.ts:107]: Implemented. `getEmailSearchUrl` now safely falls back to the default provider when `provider` is empty or unknown, so we no longer risk dereferencing an undefined config during initial account loading. Added regression coverage in `utils/url.test.ts` for both empty-string and unknown provider cases. Verified with `cd apps/web && pnpm test --run utils/url.test.ts`.

File: apps/web/utils/url.ts [hunk 2]
```diff
@@ -78,9 +91,17 @@ export function getGmailUrl(
...
+  const config = getProviderConfig("google");
+  return config.buildSearchUrl(from, emailAddress);
+}
+
+export function getEmailSearchUrl(
+  from: string,
+  emailAddress?: string | null,
+  provider?: string,
+) {
+  const config = getProviderConfig(provider);
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/utils/url.ts]: Addressed in the latest commit. `getEmailSearchUrl` now guards provider lookup and falls back to the default search URL builder when the provider is empty/unknown, preventing runtime errors from `config.buildSearchUrl`. Also added tests for empty and unknown provider fallback in `utils/url.test.ts`; targeted test run passes.
